File: backend/employees/src/main/java/com/employees/employees/application/validation/SupportedDateFormats.java
package com.employees.employees.application.validation;

import java.util.List;

public final class SupportedDateFormats {
    private static final List<String> supportedDateFormats = List.of(
            "yyyy-MM-dd",
            "dd/MM/yyyy",
            "MM/dd/yyyy",
            "dd-MM-yyyy",
            "MM-dd-yyyy",
            "dd.MM.yyyy",
            "yyyy/MM/dd",
            "yyyy.MM.dd"
    );

    public static boolean isSupported(String dateFormat) {
        if (dateFormat == null) {
            return false;
        }

        return supportedDateFormats.contains(dateFormat);
    }
}

File: frontend/employees/src/app/model/supported-date-formats.ts
import { DateFormat } from './date-format';

export const supportedDateFormats:  Array<DateFormat> = [
  { label: 'yyyy-MM-dd', example: '2025-07-17' },
  { label: 'dd/MM/yyyy', example: '17/07/2025' },
  { label: 'MM/dd/yyyy', example: '07/17/2025' },
  { label: 'dd-MM-yyyy', example: '17-07-2025' },
  { label: 'MM-dd-yyyy', example: '07-17-2025' },
  { label: 'dd.MM.yyyy', example: '17.07.2025' },
  { label: 'yyyy/MM/dd', example: '2025/07/17' },
  { label: 'yyyy.MM.dd', example: '2025.07.17' }
];
